plugins/inputs/opcua_client/opcua_client.go

```diff
@@ -0,0 +1,408 @@
+package opcua_client
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"time"
+
+	"github.com/gopcua/opcua"
+	"github.com/gopcua/opcua/ua"
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+// OpcUA type
+type OpcUA struct {
+	Name           string   `toml:"name"`
+	Endpoint       string   `toml:"endpoint"`
+	SecurityPolicy string   `toml:"security_policy"`
+	SecurityMode   string   `toml:"security_mode"`
+	Certificate    string   `toml:"certificate"`
+	PrivateKey     string   `toml:"private_key"`
+	Interval       string   `toml:"time_interval"`
+	TimeOut        int      `toml:"timeout"`
+	NodeList       []OPCTag `toml:"nodes"`
+	Nodes          []string
+	NodeData       []OPCData
+	NodeIDs        []*ua.NodeID
+	NodeIDerror    []error
+	state          ConnectionState
```

Comment:
There's no reason to have a connection state, it's just complicating the code. You can use `o.client == nil` to determine if you need to try to connect.